test/remote/gateways/remote_ipg_test.rb

```diff
@@ -20,11 +20,11 @@ def test_successful_purchase
 
   def test_successful_purchase_with_store
     payment_token = generate_unique_id
-    response = @gateway.store(@credit_card, @options.merge!({ hosted_data_id: payment_token }))
+    response = @gateway.store(@credit_card, @options.merge({ hosted_data_id: payment_token }))
```

Comment:
This is a good test, but we ALSO want it to be possible for a user to call `.store` on this gateway without passing in a `hosted_data_id`.

One problem here is that it looks to me like the gateway requires us to pass `hosted_data_id` on the store request, and even worse is that the value of `hosted_data_id` does not come back in the response. This means that the end user needs to know what `hosted_data_id` is prior to vaulting the card at the gateway with `store`.

I think we need to take some time to investigate the best way to approach this scenario.

Reply:
Done. Please review now :)